client/components/plans/plans-compare/index.jsx

```diff
@@ -151,78 +134,179 @@ var PlansCompare = React.createClass( {
 
-		plansColumns = times( numberOfPlaceholders, function( i ) {
-			var planFeatures,
-				classes = {
-					'plan-feature-column': true,
-					'is-placeholder': true,
-					'feature-list': i === 0,
-					'plan-features': i > 0
-				};
+		return (
+			<tr>{ planElements }</tr>
+		);
+	},
+
+	getTableFeatureRows: function() {
+		var plans = this.getPlans(),
+			features = this.getFeatures(),
+			rows = features.map( function( feature ) {
+				var planFeatures = plans.map( function( plan ) {
+					var content;
+
+					if ( typeof feature[ plan.product_id ] === 'boolean' && feature[ plan.product_id ] ) {
+						content = <Gridicon icon="checkmark-circle" size={ 24 } />;
+					}
+
+					if ( typeof feature[ plan.product_id ] === 'string' ) {
+						content = feature[ plan.product_id ];
+					}
+
+					return (
+						<td
+							className="plans-compare__cell is-plan-specific"
+							key={ plan.product_id }>
+							{ content }
+						</td>
+					);
+				} );
 
-			planFeatures = times( 5, function( j ) {
 				return (
-					<PlanFeatureCell key={ 'cell-' + i + '-' + j }>
-						<span></span>
-					</PlanFeatureCell>
+					<tr className="plans-compare__row" key={ feature.title }>
+						<td
+							className="plans-compare__cell"
+							key={ feature.title }>
+							{ feature.title }
+							{ this.freeTrialExceptionMarker( feature ) }
+						</td>
+						{ planFeatures }
+					</tr>
 				);
-			} );
+			}.bind( this ) );
+
+		return rows;
+	},
+
+	getTableActionRow: function() {
+		var plans = this.getPlans(),
+			cells = [ <td /> ];
 
+		cells = cells.concat( plans.map( function( plan ) {
+			var sitePlan = this.getSitePlan( plan );
 			return (
-				<div className={ classNames( classes ) } key={ 'column-' + i }>
-					<PlanHeader isPlaceholder={ true }/>
-					{ planFeatures }
-				</div>
+				<td key={ plan.product_id }>
+					<PlanActions
+						enableFreeTrials={ this.props.enableFreeTrials }
+						onSelectPlan={ this.props.onSelectPlan }
+						isInSignup={ this.props.isInSignup }
+						plan={ plan }
+						site={ this.props.selectedSite }
+						sitePlan={ sitePlan }
+						cart={ this.props.cart } />
+				</td>
 			);
+		}.bind( this ) ) );
+
+		return (
+			<tr>{ cells }</tr>
+		);
+	},
+
+	comparisonTable: function() {
+		if ( this.isDataLoading() ) {
+			return 'loading...';
+		}
+
+		return (
+			<table className="plans-compare__table">
+				<thead>
+					{ this.getTableHeader() }
+				</thead>
+				<tbody>
+					{ this.getTableFeatureRows() }
+					{ this.getTableActionRow() }
+				</tbody>
+			</table>
+		);
+	},
+
+	setPlan: function( plan ) {
+		this.setState( {
+			'currentPlan': plan
 		} );
+	},
 
+	sectionNavigationForMobile() {
 		return (
-			<div className="plans-compare">
-				<div className="plans-compare__columns">
-					{ plansColumns }
-				</div>
-			</div>
+			<span className="plans-compare__section-navigation">
+				<SectionNav>
+					<NavTabs>
+						<NavItem onClick={ this.setPlan.bind( this, 'free' ) }>
```

Comment:
@scruffian One of our linting rules is to not call `Function.prototype.bind` in JSX (basically, in render()) because it creates a new function each time there is a render.